File: CS50_2020/Week1/Cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // prompt the amount of money given by customer
    float amount;
    do
    {
        amount = get_float("Change owned: ");
    }
    while (amount <= 0);
    
    // convert dolars to cents
    int coins = round(amount * 100);
    
    int quaters = coins / 25;
    int dimes = (coins % 25) / 10;
    int nickels = ((coins % 25) % 10) / 5;
    int pennies = ((coins % 25) % 10) % 5;
    int total = quaters + dimes + nickels + pennies;
    printf("%i\n", total);
}

File: CS50_2020/Week6/Cash/cash.py
# use get_float from cs50 library
from cs50 import get_float

# initialize for user input
validate = False

# check if user input is correct
while not validate:
    cash = get_float("Change owed: ")
    if cash > 0:
        validate = True
    # convert to cents
    amount = cash * 100

# number of coint possible
quaters = amount // 25
dimes = (amount % 25) // 10
nickels = ((amount % 25) % 10) // 5
pennies = ((amount % 25) % 10) % 5
total = quaters + dimes + nickels + pennies

print(int(total))